app-backend/db/src/main/scala/ProjectDao.scala

```diff
@@ -245,6 +245,15 @@ object ProjectDao
     }
   }
 
+  def getShareCount(projectId: UUID, userId: String): ConnectionIO[Long] = {
+    val sharedUsers = ProjectDao.getPermissions(projectId).map { acrList =>
+      acrList.flatten.filter(_.subjectType == SubjectType.User).flatMap { acr =>
```

Comment:
I think you can make this a little bit more straightforward with a `collect` -- https://www.scala-lang.org/api/2.12.8/scala/collection/immutable/List.html#collect[B](pf:PartialFunction[A,B]):List[B]:

```scala
acrList.collect {
  case Some(acr@ObjectAccessControlRule(subjType, _, _)) if subjType == SubjectType.User => acr
}
```

I _think_ you're allowed to decorate inside an unapply at least

Reply:
yeah, this worked -- I was actually able to match on `Some(subjectId)` too that got rid of the `Option` there. Good suggestion.

Resolved in https://github.com/raster-foundry/raster-foundry/pull/5292/commits/d98fee4ae7251cad26ad31a9106219692ce789cd